torchvision/prototype/transforms/_utils.py

```diff
@@ -39,14 +39,34 @@ def get_image_dimensions(image: Union[PIL.Image.Image, torch.Tensor, features.Im
     return channels, height, width
 
 
-def has_any(sample: Any, *types: Type) -> bool:
+def has_any(sample: Any, *types_or_checks: Union[Type, Callable[[Any], bool]]) -> bool:
     flat_sample, _ = tree_flatten(sample)
-    return any(issubclass(type(obj), types) for obj in flat_sample)
+    for type_or_check in types_or_checks:
+        passed_check = False
+        for obj in flat_sample:
+            if isinstance(obj, type_or_check) if isinstance(type_or_check, type) else type_or_check(obj):
+                passed_check = True
+                break
 
+        if passed_check:
+            return True
```

Comment:
Nit: Isn't this the same as:
```suggestion
        for obj in flat_sample:
            if type_or_check(obj) if callable(type_or_check) else isinstance(obj, type_or_check):
                return True
```